### ISSUE TITLE
pi-GAN: Periodic Implicit Generative Adversarial Networks for 3D-Aware Image Synthesis

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#78]
*CVPR 2021 Oral*

## Motivation

- 3D-aware Generative Model (View Consistency)
- Higher Image Quality

## Key Idea

- SIREN network (sine activation)
- NeRF Representation (+ Differentiable Volume Rendering)
- Latent Vector conditioning via Mapping Network (sine activation frequency and phase shift - AdaIN style)
- Progressive Growing training scheme

## Previous Approach

- HoloGAN : voxel feature representation - susceptible to huge view change
- GRAF : NeRF feature representation - not expressive enough for fine details (MLP + Positional Encoding)

### Overview
![image](https://user-images.githubusercontent.com/46308407/127250954-5d3ba0db-6b55-4fe0-8208-71314ca8d370.png)

- We query position (x,y,z) into SIREN MLP which conditions on mapped latent vector.
- Mapping network is a leaky-ReLU MLP that maps latent vector z to beta and gamma.
- Beta and gamma are parameters for sine activation functions in SIREN MLP (phase shift and frequency)
- We render a 2D projection and discriminator distinguishes it with real image.
- Note that volume density **does not** depend on camera view point.

### Objective
![image](https://user-images.githubusercontent.com/46308407/127251177-cda67bef-8fe7-4fcc-92e8-70970966e59a.png)

- Non-saturating GAN loss with R1 regularization.
- Patch discriminator is intentionally avoided (unlike GRAF) to evade severe overfitting.

### Progressive Training Scheme

- Discriminator gets additional layers twice in training (32x32 -> 64x64 -> 128x128)
- Generator is not changed, but **ray sampling density** is elevated. (Higher resolution)
- This enables larger batch size in earlier training (low resolution regime) -> faster and more stable.

### Visual Inspection

![image](https://user-images.githubusercontent.com/46308407/127251425-57eaf096-3ca5-4840-bdcd-43c136427f53.png)
![image](https://user-images.githubusercontent.com/46308407/127251474-2bfd96d9-5ee7-4b5a-ac8e-562c524d075c.png)

### Quantitative Comparison
![image](https://user-images.githubusercontent.com/46308407/127251669-155eb2c3-02bd-4359-a59e-c4dbbe1987fc.png)

### Ablation
![image](https://user-images.githubusercontent.com/46308407/127251720-fe063528-127b-4631-91a9-268fe1c2792b.png)
![image](https://user-images.githubusercontent.com/46308407/127251726-810e4120-c06e-4bca-92ac-c59eb8708241.png)

### Analysis
![image](https://user-images.githubusercontent.com/46308407/127251745-7be50cfc-7c15-41f5-b12f-4556a918b4cc.png)
![image](https://user-images.githubusercontent.com/46308407/127251750-353d6d94-8713-4b51-92f1-2212725e6451.png)
![image](https://user-images.githubusercontent.com/46308407/127251759-59562705-e9bb-4862-9408-11e9168b3292.png)

### Application
![image](https://user-images.githubusercontent.com/46308407/127251779-c19f08c8-26f8-45d1-a6e5-4df4a340fc19.png)
![image](https://user-images.githubusercontent.com/46308407/127251789-1d63aea0-d0d1-4d6b-9286-8f9bae20ccba.png)

## Comments
- Very intuitive and easy to read.
- Sinusoidal activation is more expressive than ReLU with Fourier features.
- Implicit 3D generative model can be trained without pixel-wise loss guidance.
- I know it's easy to use rule-based volume rendering, but wouldn't it be the next step to *learn* it with data?
- For now, black box projections suffer from view inconsistency and susceptibility to OOD camera poses, but one will come up with reasonable remedies.
- Also, why not alter focal length? It would be cool if they showed us *out-focused* samples. Is it not possible for now?
